api/db-migrations/000013_add_default_traffic_rule.up.sql

```diff
@@ -0,0 +1,78 @@
+ALTER TABLE router_versions ADD default_traffic_rule jsonb;
+
+WITH cleaned_router_versions AS (
+    SELECT id, CASE WHEN traffic_rules = 'null' THEN '[]'::jsonb ELSE traffic_rules END AS traffic_rules
+    FROM router_versions
+),
+rule_tbl AS (
+    SELECT id, jsonb_array_elements(traffic_rules) AS rule FROM cleaned_router_versions
+),
+-- explode the route names from each rule
+rule_route_tbl AS (
+    SELECT id, jsonb_array_elements(rule->'routes') AS routes FROM rule_tbl
+),
+-- explode the original routes column
+route_tbl AS (
+    SELECT id, jsonb_array_elements(routes) AS route FROM router_versions
+),
+original_route_tbl AS (
+    SELECT id, route -> 'id' AS route_name FROM route_tbl
+),
+dangling_routes AS (
+    SELECT
+        id, jsonb_agg(route_name) dangling_routes
+    FROM (
+        SELECT * FROM original_route_tbl
+        EXCEPT
+        SELECT * FROM rule_route_tbl
+    ) t
+    GROUP BY t.id
+),
+exploded_traffic_rules AS (
+    SELECT
+        id
+        , position
+        , elem
+    FROM cleaned_router_versions, jsonb_array_elements(traffic_rules) WITH ordinality arr(elem, position)
+),
+-- Need to join from cleaned_router_versions since exploded_traffic_rules will exclude router versions which have no traffic rules
```

Comment:
Why do we care about router versions with no traffic rules?

Reply:
Overall the `up` query looks ok but perhaps we can further simplify this by focussing only on the records with traffic rules since that's our goal for updating the traffic_rules as well as default_traffic_rule. Could you try that?

I'll have one last look at the up & down sql scripts after this.

Reply:
Yeah I think we can simplify if we focus only on the records with traffic rules - I'll test it on internal staging DB records as well.